chapter01/worksheets/solution1.6.2.5.scala

```diff
@@ -0,0 +1,24 @@
+2/** author: Andreas Röhler */
+
+/**
+  Exercise 1.6.2.5
+  Define a function of type List[Double] => List[Double] that “normalizes” the list:
+
+  it finds the element having the largest absolute value and, if that
+value is nonzero, divides all elements by that value and returns a new
+  list; otherwise returns the original list.
+
+  Test with: scala> normalize(List(1.0, -4.0, 2.0))
```

Comment:
You put `-4.0` here, while in the book it says `4.0`.

This raises an interesting question: what should the function do if the largest value is a negative number?
- divide by that negative number
- divide by the absolute value of that negative number

Your solution takes the absolute value of all elements in the list. I'm not sure this is quite clear why that is better.

Let's change the expected test to:

```scala
val result = normalize(List(1.0, -4.0, 2.0))
val expected: List[Double] = List(-0.25, 1.0, -0.5)

assert(result == expected) 
```

Reply:
Hmm, the exercise speaks of "largest absolute value". Concluded, there might be negative numbers too. See also https://github.com/winitzki/sofp/issues/84

Reply:
Right. So for List(1.0, -4.0, 2.0) the largest absolute value is 4. But the list element with the largest absolute value is -4.0.

I'll have to rewrite the text to make it clear.

Let's revert to the following:
```scala
val result = normalize(List(1.0, -4.0, 2.0))
val expected: List[Double] = List(0.25, -1.0, 0.5)

assert(result == expected) 
```

Reply:
Okay, sorry being that dumb. Understand the text now, no changes required. It's about 'all elements', not abs-values of all elements.